api/tests/unit/domain/usecases/find-available-target-profiles_test.js

```diff
@@ -0,0 +1,63 @@
+const usecases = require('../../../../lib/domain/usecases');
+const { expect, sinon, factory } = require('../../../test-helper');
+const TargetProfile = require('../../../../lib/domain/models/TargetProfile');
+
+describe('Unit | UseCase | find-available-target-profiles', () => {
+
+  let sandbox;
+  const targetProfileRepository = {
+    findByFilters: () => undefined
+  };
+  let organizationId;
+  let targetProfilesLinkedToOrganization;
+  let publicTargetProfiles;
+
+  beforeEach(() => {
+    organizationId = 1;
+    targetProfilesLinkedToOrganization = [factory.buildTargetProfile({ organizationId })];
+    publicTargetProfiles = [factory.buildTargetProfile({ isPublic: true })];
+
+    sandbox = sinon.sandbox.create();
+    sandbox.stub(targetProfileRepository, 'findByFilters');
+    targetProfileRepository.findByFilters.withArgs({ isPublic: true }).resolves(publicTargetProfiles);
+    targetProfileRepository.findByFilters.withArgs({ organizationId }).resolves(targetProfilesLinkedToOrganization);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('should return an array of target profiles', () => {
+    // when
+    const promise = usecases.findAvailableTargetProfiles({ organizationId, targetProfileRepository });
+
+    // then
+    return promise.then((availableTargetProfiles) => {
+      expect(availableTargetProfiles).to.be.an('array');
+      expect(availableTargetProfiles[0]).to.be.an.instanceOf(TargetProfile);
+    });
+  });
+
+  it('should find public profiles and profiles linked to anyOrganization', () => {
+    // when
+    const promise = usecases.findAvailableTargetProfiles({ organizationId, targetProfileRepository });
+
+    // then
+    return promise.then(() => {
+      expect(targetProfileRepository.findByFilters).to.have.been.calledTwice;
+      expect(targetProfileRepository.findByFilters).to.have.been.calledWith({ organizationId: organizationId });
+      expect(targetProfileRepository.findByFilters).to.have.been.calledWith({ isPublic: true });
+    });
+  });
+
+  it('should return public profiles and profiles linked to specified organization', () => {
+    // when
+    const promise = usecases.findAvailableTargetProfiles({ organizationId, targetProfileRepository });
+
+    // then
+    return promise.then((availableTargetProfiles) => {
+      expect(availableTargetProfiles.length).to.equal(2);
```

Comment:
J'aurais plutôt fait un `deep.equal` pour être sur de ce que l'on me renvoie.

Reply:
to.include.deep.members :D

Reply:
Done